src/transformers/modeling_utils.py

```diff
@@ -1829,10 +1851,12 @@ def save_pretrained(
                 UserWarning,
             )
 
-        if getattr(self, "is_loaded_in_4bit", False):
-            raise NotImplementedError(
-                "You are calling `save_pretrained` on a 4-bit converted model. This is currently not supported"
-            )
+        if getattr(self, "is_loaded_in_4bit", False) and getattr(self, "is_4bit_serializable", False):
+            warnings.warn(
+                "You are calling `save_pretrained` to a 4-bit converted model you may likely encounter unexepected"
+                " behaviors. If you want to save 4-bit models, make sure to have `bitsandbytes>=0.42` installed.",
+                UserWarning,
+            )  # TODO:  verify version
```

Comment:
I think we should raise an error here and not allow users to save 4bit models if they have the wrong bnb version - the condition that we have put for 8bit is a bit too flexible

Reply:
done. changed to `NotImplementedError` for both 8 and 4 bits.

Reply:
Nice, great !